pkg/apb/ext_creds.go

```diff
@@ -24,29 +24,46 @@ import (
 	"strings"
 	"time"
 
+	"github.com/openshift/ansible-service-broker/pkg/clients"
 	"github.com/openshift/ansible-service-broker/pkg/runtime"
+	"github.com/openshift/ansible-service-broker/pkg/version"
 
 	logging "github.com/op/go-logging"
 )
 
+type extractCreds func(
+	string,
+	string,
+	*logging.Logger,
+	*clients.KubernetesClient,
+) (*ExtractedCredentials, error)
+
 // ExtractCredentials - Extract credentials from pod in a certain namespace.
 func ExtractCredentials(
-	podname string, namespace string, log *logging.Logger,
+	podname string,
+	namespace string,
+	runtimeVersion int,
+	log *logging.Logger,
 ) (*ExtractedCredentials, error) {
-	log.Debug("Calling monitorOutput on " + podname)
-	bindOutput, err := monitorOutput(namespace, podname, log)
+	k8s, err := clients.Kubernetes(log)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Unable to retrive kubernetes client - %v", err)
 	}
 
-	if bindOutput == nil {
-		return nil, nil
+	extractCredsFunc, err := getExtractCreds(runtimeVersion)
+	if err != nil {
+		return nil, err
 	}
-
-	return buildExtractedCredentials(bindOutput)
+	return extractCredsFunc(podname, namespace, log, k8s)
 }
 
-func monitorOutput(namespace string, podname string, log *logging.Logger) ([]byte, error) {
+// ExtractCredentialsAsFile - Extract credentials from running APB using exec
+func ExtractCredentialsAsFile(
+	podname string,
+	namespace string,
+	log *logging.Logger,
+	k8s *clients.KubernetesClient,
```

Comment:
Why are we passing in the k8s client? I think the right thing to do here is getting it from the method.

Reply:
Fair point.

Reply:
I understand this to mean why pass it when you can just ask for another one with `k8s, err := clients.Kubernetes(log)`. Is that right?

Reply:
yes, that is exactly what I was thinking

Reply:
I hate it that I agree with you b/c then I have to change it. :sunglasses:

Reply:
I'm going to wait to push this change until there are thoughts on the method declarations.

Reply:
I'm fine with either.  I leave it up to the writer to choose as long as the function wouldn't normally fit on one line.

Reply:
I don't think we need to block on this since we haven't really established a style here. And we would have to change it everywhere to match the style we go with.  We can discuss in a separate issue if we want to pursue it.